core/unit_test/TestTeamVector.hpp

```diff
@@ -605,6 +605,71 @@ struct functor_vec_scan {
   }
 };
 
+// Temporary: This condition will progressively be reduced when parallel_scan
+// with return value will be implemented for more backends.
+#if defined(KOKKOS_ENABLE_SERIAL) || defined(KOKKOS_ENABLE_OPENMP)
+#if !defined(KOKKOS_ENABLE_CUDA) && !defined(KOKKOS_ENABLE_HIP) &&             \
+    !defined(KOKKOS_ENABLE_OPENACC) && !defined(KOKKOS_ENABLE_SYCL) &&         \
+    !defined(KOKKOS_ENABLE_THREADS) && !defined(KOKKOS_ENABLE_OPENMPTARGET) && \
+    !defined(KOKKOS_ENABLE_HPX)
+template <typename Scalar, class ExecutionSpace>
+struct functor_vec_scan_ret_val {
+  using policy_type     = Kokkos::TeamPolicy<ExecutionSpace>;
+  using execution_space = ExecutionSpace;
+  using view_type       = Kokkos::View<Scalar **, execution_space>;
+
+  Kokkos::View<int, Kokkos::LayoutLeft, ExecutionSpace> flag;
+  view_type return_values;
+  int team_size;
+
+  functor_vec_scan_ret_val(
+      Kokkos::View<int, Kokkos::LayoutLeft, ExecutionSpace> flag_, int nteams,
+      int tsize)
+      : flag(flag_), team_size(tsize) {
+    return_values = view_type("return_values", nteams, tsize);
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(typename policy_type::member_type team) const {
+    Scalar return_val;
+
+    Kokkos::parallel_scan(
+        Kokkos::ThreadVectorRange(team, 13),
+        [&](int i, Scalar &val, bool final) {
+          val += i;
+
+          if (final) {
+            Scalar test = 0;
+            for (int k = 0; k <= i; k++) test += k;
+            return_values(team.league_rank(), team.team_rank()) = test;
+
+            if (test != val) {
+              Kokkos::printf("FAILED vector_par_scan %i %i %lf %lf\n",
+                             team.league_rank(), team.team_rank(),
+                             static_cast<double>(test),
+                             static_cast<double>(val));
+
+              flag() = 1;
+            }
+          }
```

Comment:
Mixing computation and checking the result is not good practice.

Reply:
btw this is following the similar approach used in the other tests here, so if you want to change the overall approach maybe we can do that in a separate PR?